packages/shared/src/features/onboarding/steps/FunnelRegistration.tsx

```diff
@@ -104,35 +99,69 @@ const useRegistrationListeners = (
   useEventListener(globalThis, 'message', onProviderMessage);
 };
 
-export function FunnelRegistration({
-  heading,
-  image,
-  imageMobile,
+const checkIsInAppAndroid = () => {
+  const inAppBrowser = isWebView();
+  const confirmedIOS = isIOS();
+
+  return inAppBrowser && !confirmedIOS;
+};
+
+function InnerFunnelRegistration({
+  parameters: { headline, image, imageMobile },
   onTransition,
-}: FunnelRegistrationProps): ReactElement {
+}: FunnelStepSignup): ReactElement {
+  const router = useRouter();
   const isTablet = useViewSize(ViewSize.Tablet);
+  const isAndroidWebView = checkIsInAppAndroid();
+  const config = useMemo(() => {
+    if (!router?.isReady) {
+      return { enabled: false };
+    }
+
+    if (!isAndroidWebView) {
+      return { enabled: true };
+    }
+
+    if (typeof window === 'undefined') {
+      return { enabled: false };
+    }
+
+    return { enabled: true, redirect_to: window.location.href };
```

Comment:
Are we sure this workaround is needed only for android webview?

Reply:
@sshanzel mostly for you i guess

Reply:
Haven't checked iOS webview, but we can also apply it if that has never worked in iOS since then.

Reply:
Ok we will need to test it on ios as well. let's leave it as it is for now

Reply:
Just checked FB browser on iOS, doesn't work too. Raising a PR.

Reply:
I also tested a few platforms, it is as if Facebook is the only broken one 👀

Reply:
I think it would be difficult to audit and maintain accurate assumption whether it would work from these webviews. We can probably just use the redirection when it is webview regardless if it works with window.open in the past.

![Screenshot 2025-04-12 at 5 10 21 AM](https://github.com/user-attachments/assets/54cd87f3-bdef-4d9e-98bf-34789e45574b)

Reply:
Is there a better way to detect in-app? Can we detect if window.open doesn't work somehow?